External/FEXCore/Source/Interface/Core/CPUID.cpp

```diff
@@ -25,89 +28,321 @@ CPUIDEmu::FunctionResults CPUIDEmu::Function_0h() {
   CPUIDEmu::FunctionResults Res{};
-  Res.Res[0] = 0x8000001F;
+  Res.eax = 0x8000001F;
   return Res;
 }
 
 // Extended processor and feature bits
 CPUIDEmu::FunctionResults CPUIDEmu::Function_8000_0001h() {
   CPUIDEmu::FunctionResults Res{};
-  Res.Res[2] = ~0U; // Let's say we support every feature for fun
-  Res.Res[3] = ~0U; // Let's say we support every feature for fun
-
-  Res.Res[3] &= ~(
-    (1 << 6)  | // Remove SSE4a
-    (1 << 11) | // Remove XOP
-    (1 << 16)   // Remove FMA4
-    );
+
+  Res.eax = 0 | // Stepping
+    (0 << 4) | // Model
+    (0 << 8) | // Family ID
+    (0 << 12) | // Processor type
+    (0 << 16) | // Extended model ID
+    (0 << 20); // Extended family ID
+
+  Res.ecx =
+    (1 <<  0) | // LAHF/SAHF
+    (1 <<  1) | // 0 = Single core product, 1 = multi core product
+    (0 <<  2) | // SVM
+    (1 <<  3) | // Extended APIC register space
+    (0 <<  4) | // LOCK MOV CR0 means MOV CR8
+    (1 <<  5) | // ABM instructions
+    (0 <<  6) | // SSE4a
+    (0 <<  7) | // Misaligned SSE mode
+    (1 <<  8) | // PREFETCHW
+    (0 <<  9) | // OS visible workaround support
+    (0 << 10) | // Instruction based sampling support
+    (0 << 11) | // XOP
+    (0 << 12) | // SKINIT
+    (0 << 13) | // Watchdog timer support
+    (0 << 14) | // Reserved
+    (0 << 15) | // Lightweight profiling support
+    (0 << 16) | // FMA4
+    (1 << 17) | // Translation cache extension
+    (0 << 18) | // Reserved
+    (0 << 19) | // Reserved
+    (0 << 20) | // Reserved
+    (0 << 21) | // Reserved
+    (1 << 22) | // Topology extensions support
+    (1 << 23) | // Core performance counter extensions
+    (1 << 24) | // NB performance counter extensions
+    (0 << 25) | // Reserved
+    (0 << 26) | // Data breakpoints extensions
+    (1 << 27) | // Performance TSC
+    (1 << 28) | // L2 perf counter extensions
+    (0 << 29) | // Reserved
+    (0 << 30) | // Reserved
+    (0 << 31);  // Reserved
+
+  Res.edx =
+    (1 <<  0) | // FPU
+    (1 <<  1) | // Virtual mode extensions
+    (1 <<  2) | // Debugging extensions
+    (1 <<  3) | // Page size extensions
+    (1 <<  4) | // TSC
+    (1 <<  5) | // MSR support
+    (1 <<  6) | // PAE
+    (1 <<  7) | // Machine Check Exception
+    (1 <<  8) | // CMPXCHG8B
+    (1 <<  9) | // APIC
+    (0 << 10) | // Reserved
+    (1 << 11) | // SYSCALL/SYSRET
+    (1 << 12) | // MTRR
+    (1 << 13) | // Page global extension
+    (1 << 14) | // Machine Check architecture
+    (1 << 15) | // CMOV
+    (1 << 16) | // Page attribute table
+    (1 << 17) | // Page-size extensions
+    (0 << 18) | // Reserved
+    (0 << 19) | // Reserved
+    (1 << 20) | // NX
+    (0 << 21) | // Reserved
+    (1 << 22) | // MMXExt
+    (1 << 23) | // MMX
+    (1 << 24) | // FXSAVE/FXRSTOR
+    (1 << 25) | // FXSAVE/FXRSTOR Optimizations
+    (1 << 26) | // 1 gigabit pages
+    (0 << 27) | // RDTSCP
+    (0 << 28) | // Reserved
+    (1 << 29) | // Long Mode
+    (1 << 30) | // 3DNow! Extensions
+    (1 << 31);  // 3DNow!
```

Comment:
Do we really want to claim to support **3DNow!**

Reply:
Disabled